splunk-otel-android/src/main/java/com/splunk/rum/SplunkRumBuilder.java

```diff
@@ -0,0 +1,323 @@
+        return this;
+    }
+
+    /**
+     * Sets the "beacon" endpoint URL to be used by the RUM library.
+     *
+     * <p>Note that if you are using standard Splunk ingest, it is simpler to just use {@link
+     * #setRealm(String)} and let this configuration set the full endpoint URL for you.
+     *
+     * @return {@code this}
+     */
+    public SplunkRumBuilder setBeaconEndpoint(String beaconEndpoint) {
+        if (realm != null) {
+            Log.w(
+                    SplunkRum.LOG_TAG,
+                    "Explicitly setting the beaconEndpoint will override the realm configuration.");
+            realm = null;
+        }
+        this.beaconEndpoint = beaconEndpoint;
+        return this;
+    }
+
+    /**
+     * Sets the realm for the beacon to send RUM telemetry to. This should be used in place of the
+     * {@link #setBeaconEndpoint(String)} method in most cases.
+     *
+     * @param realm A valid Splunk "realm"
+     * @return {@code this}
+     */
+    public SplunkRumBuilder setRealm(String realm) {
+        if (beaconEndpoint != null && this.realm == null) {
+            Log.w(
+                    SplunkRum.LOG_TAG,
+                    "beaconEndpoint has already been set. Realm configuration will be ignored.");
+            return this;
+        }
+        this.beaconEndpoint = "https://rum-ingest." + realm + ".signalfx.com/v1/rum";
+        this.realm = realm;
+        return this;
+    }
+
+    /**
+     * Sets the RUM auth token to be used by the RUM library.
+     *
+     * @return {@code this}
+     */
+    public SplunkRumBuilder setRumAccessToken(String rumAuthToken) {
+        this.rumAccessToken = rumAuthToken;
+        return this;
+    }
+
+    /**
+     * Enables debugging information to be emitted from the RUM library.
+     *
+     * <p>This feature is disabled by default. You can enable it by calling this configuration
+     * method with a {@code true} value.
+     *
+     * @return {@code this}
+     */
+    public SplunkRumBuilder enableDebug(boolean enabled) {
+        this.debugEnabled = enabled;
+        return this;
+    }
+
+    /**
+     * Enables the storage-based buffering of telemetry. If this feature is enabled, telemetry is
+     * buffered in the local storage until it is exported; otherwise, it is buffered in memory and
+     * throttled.
+     *
+     * <p>This feature is disabled by default. You can enable it by calling this configuration
+     * method with a {@code true} value.
+     *
+     * @return {@code this}
+     */
+    public SplunkRumBuilder enableDiskBuffering(boolean enabled) {
+        this.diskBufferingEnabled = enabled;
+        return this;
+    }
+
+    /**
+     * Disables the crash reporting feature.
+     *
+     * <p>This feature is enabled by default. You can disable it by calling this configuration
+     * method with a {@code true} value.
+     *
+     * @return {@code this}
+     */
+    public SplunkRumBuilder disableCrashReporting(boolean disabled) {
```

Comment:
I decided to rename these boolean settings methods a bit, I used https://github.com/open-telemetry/opentelemetry-specification/issues/2729#issuecomment-1222584006 as the inspiration -- the default value of all of these methods is `false`.

I was wondering whether we need the boolean parameters at all, after all we could just assume `true` -- but I think that in the beginning of this project we decided to accept a boolean arg to make setting these flags based on configuration easier.

Reply:
> Make sense?

Yeah 👍 

> for users who might have to pass the builder around a few places and might want to override a previous components setting on the builder, the parity is nice.

I think that we could possibly add setter methods (`setDebugEnabled(boolean)`) for these scenarios; but I think I'd rather skip them for now and only add them when we're requested to.